### PR TITLE
zpool-auto-expand-partitions: init at 0.1.0, update auto-expand service

### PR DESCRIPTION
**Pull request by grahamc**
###### Description of changes

Replace the not-very-good auto expand service we have with zpool-auto-expand-partitions, which will also auto-expand underlying partitions.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/tasks/filesystems/zfs.nix [hunk 1]
```diff
@@ -642,41 +642,14 @@ in
...
-        # See: https://github.com/openzfs/zfs/issues/12505
-        script = let
-          # This UUID has been chosen at random and is to provide a
-          # collision-proof, predictable token to search for
-          magicIdentifier = "NIXOS-ZFS-ZPOOL-DEVICE-IDENTIFIER-37108bec-aff6-4b58-9e5e-53c7c9766f05";
-          zpoolScripts = pkgs.writeShellScriptBin "device-highlighter" ''
-            echo "${magicIdentifier}"
-          '';
-        in ''
+        path = [ pkgs.gawk cfgZfs.package pkgs.parted pkgs.cloud-utils pkgs.util-linux ];
```
Comment (cole-h) [inline nixos/modules/tasks/filesystems/zfs.nix:645]: (Missed this earlier)

Do we really need `cloud-utils` and `util-linux` in the path if we `substituteInPlace`d them in the actual `zpool-auto-expand-partitions` derivation?

Should we also substitute `parted` in that derivation?

Comment (SuperSandro2000) [reply nixos/modules/tasks/filesystems/zfs.nix:645]:
```suggestion
        path = with pkgs; [ gawk cfgZfs.package parted cloud-utils util-linux ];
```
I always find this easier to parse the packages.

**Review by cole-h — APPROVED** [review]
Looks much better -- and I also like the cleanup of the `path` list.

File: pkgs/tools/filesystems/zpool-auto-expand-partitions/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  src = fetchFromGitHub {
+    owner = "DeterminateSystems";
+    repo = "zpool-auto-expand-partitions";
+    rev = "v${version}";
+    hash = "sha256-LA6YO6vv7VCXwFfayQVxVR80niSCo89sG0hqh0wDEh8=";
+  };
+
+  cargoHash = "sha256-5v0fqp8aro+QD/f5VudMREc8RvKQapNAoArcCKMN1Sw=";
+
+  preBuild = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/filesystems/zpool-auto-expand-partitions/default.nix:23]: Patching should be done in postPatch.

**Pull request merged**